clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -614,6 +615,24 @@ public static String stackTrace(Throwable e) {
         return Utils.readBytes(buffer, 0, buffer.limit());
     }
 
+    /**
+     * Read a file as string and return the content. Unlike {@link Utils#readFileAsString(String)}, the file
+     * will be treated as a stream and a character is read at a time. This allows the program to read from say
+     * a FIFO (opened with S_IFIFO flag), where size of file is not known in advance.
+     */
+    public static String readFileStreamAsString(String path) {
+        StringBuilder fileText = new StringBuilder();
+        try (FileInputStream fileInputStream = new FileInputStream(path)) {
+            int nextChar;
+            while ((nextChar = fileInputStream.read()) != -1) {
```

Comment:
This does a system call every time. We can use `InputStream`'s `readAllBytes`: https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/io/InputStream.java#L389

Reply:
If we do this, we should have an array of `byte`s and can just call `String​(byte[] bytes, Charset charset)`.

Reply:
Unfortunately the call is not in Java 8. Thanks for suggestion, thankfully this is not in critical performance path, and the files we are reading are less than a kilobyte, so it should be fine.

Reply:
Also charset suggestion is cool. Once we drop Java 8 support, we can come back to this and improve it.

Reply:
This is a utility method so we can't say whether it's used in perf critical code or not, right?

Reply:
Current usage is all in *Command classes, but point is valid. In future someone can put this in performance critical path. I have updated based on @mumrah's suggestion to use `Files#readAllBytes` and this method reads all bytes in one go.

Now the `Files#readAllBytes` uses a SeekableByteChannel and gets the size of the file to work, but in tests it seems to be working for a named pipe which isn't seekable. I am going to look into the native code that returns size to check what is going on. If all checks out I will update the PR with commit.

Reply:
So looked into the code and the `size` method simply does fstat and returns the value (so there is no seek). Here is the code:
```
    if (fstat64(fd, &fbuf) < 0)
        return handle(env, -1, "Size failed");
...
    return fbuf.st_size;
```
For the FIFO the value returned is 0 and then the code in `Files#read` then allocates a array of 8K size and reads data in chunks of 8K. So I think we are good with the latest change that @mumrah suggested. Going to post the commit now.